query/iterator.hpp

```diff
@@ -421,13 +421,13 @@ class ArrayVectorPartyIterator
             reinterpret_cast<uint64_t *>(basePtr));
 
     if (length == 0) {
-        return thrust::make_tuple(reinterpret_cast<Value*>(NULL), true);
-    } else if (length == UINT32_MAX) {
+      if (offset == 0) {
         return thrust::make_tuple(reinterpret_cast<Value*>(NULL), false);
-    } else {
-        return thrust::make_tuple(
-                reinterpret_cast<Value*>(valuePtr + offset), true);
+      }
+      return thrust::make_tuple(reinterpret_cast<Value*>(NULL), true);
```

Comment:
can be simplified to return thrust::make_tuple(reinterpret_cast<Value*>(NULL), offset == 0);